src/main/java/com/mineboundteam/minebound/magic/network/MagicButtonSync.java

```diff
@@ -19,87 +16,75 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraftforge.network.NetworkEvent;
 
+import java.util.function.Supplier;
+
 public class MagicButtonSync {
-    
+
     public static void handleButtons(ButtonMsg msg, Supplier<NetworkEvent.Context> ctx) {
         ctx.get().enqueueWork(() -> {
             ServerPlayer player = ctx.get().getSender();
             if (!player.level.isClientSide()) {
-    
                 var activeSpells = new InventorySpellContainer(player.getInventory(), PlayerSelectedSpellsProvider.PRIMARY_SPELL, ArmorNBTHelper.ACTIVE_SPELL, true);
                 var passiveSpells = new InventorySpellContainer(player.getInventory(), PlayerSelectedSpellsProvider.PRIMARY_SPELL, ArmorNBTHelper.PASSIVE_SPELL, true);
                 boolean canOpen = !(activeSpells.isEmpty() && passiveSpells.isEmpty());
-    
+
                 ItemStack chest = player.getItemBySlot(EquipmentSlot.CHEST);
                 switch (msg.msg) {
                     case PRIMARY_MENU -> {
-                        if (canOpen && chest.getItem() instanceof MyrialArmorItem && ((MyrialArmorItem) chest.getItem()).getTier().handSlots > 0)
+                        if (canOpen && chest.getItem() instanceof MyrialArmorItem item && item.getTier().handSlots > 0)
                             player.openMenu(new SimpleMenuProvider((containerID, inventory, p) -> new SelectSpellMenu(containerID, inventory, true), TextComponent.EMPTY));
                     }
                     case PRIMARY_PRESSED -> {
-                        if (chest.getItem() instanceof MyrialArmorItem && ((MyrialArmorItem) chest.getItem()).getTier().handSlots > 0) {
+                        if (chest.getItem() instanceof MyrialArmorItem item && item.getTier().handSlots > 0) {
                             UseSpellHelper.useSpell(player, PlayerSelectedSpellsProvider.PRIMARY_SPELL);
-                            
+
                             int handStates = MagicEvents.playerStates.getOrDefault(player.getId(), 0);
                             handStates |= MagicAnimationSync.ArmUsersMsg.PRIMARY;
-                            MagicEvents.playerStates.put(player.getId(), handStates);
```

Comment:
Why did you remove the put here?

Reply:
Oops, accident from when I was debugging the arm issue. Fixed.